### ISSUE TITLE
Pattern grid cannot be scrolled horizontally

### ISSUE THREAD

**Issue by ISSOtm** [stoneface86/trackerboy#21]
## Bug description

If the tracker is resized to be too small for the pattern grid, then it's impossible to scroll the view horizontally, and the rightmost channel(s) are inaccessible. The scrollbar moves the cursor, but it goes off-screen:

![Screenshot with the cursor all the way left](https://github.com/stoneface86/trackerboy/assets/15271137/07e2c3b9-b5c7-4c55-96d4-bfd96c03a2f7)
![Screenshot with the cursor partially off-screen](https://github.com/stoneface86/trackerboy/assets/15271137/16caf187-fe15-4d9b-b47d-bcdb5b9e23b2)

## How to reproduce

1. Shrink the tracker window horizontally

## Information

 - OS: Arch Linux
 - Architecture: x86_64
 - Trackerboy version: 0.6.5 (08753de60e458f520cfabf64448ad25d90460903)

**Comment by stoneface86** [comment]
Yeah the pattern editor and header definitely need to be aware of the viewport. Currently it just assumes there's enough room to draw the whole pattern which results in clipping when that is not the case as shown by the example above.

 I think keeping the row numbers "frozen" is a good idea, such that they're always drawn. We can change it so that if say only 3 tracks can fit in the control and the cursor is on CH2 (or CH3, CH4), then only CH2, CH3 and CH4 is drawn.
